### PR TITLE
Animate Line + some minor cleanup

### PR DESCRIPTION
**Pull request by aaronmgdr**
### Description

On BrandKit pages changes the thin gray line on the bottom of top menu bar to only show when content is near it or under it. 

### Tested
Screen Shot 2020-02-21 at 12 04 58 PM
Screen Shot 2020-02-21 at 12 05 11 PM

### Other changes

fix warning about Link getting invalid href (Link should only be used for internal hrefs)

### Related issues

- Fixes https://github.com/celo-org/celo-labs/issues/224

### CODE REVIEW

File: packages/web/src/brandkit/common/Page.tsx [hunk 1]
```diff
@@ -188,27 +194,44 @@ class Page extends React.Component<Props & ScreenProps, State> {
...
-      const element = (findNodeHandle(value.current) as unknown) as Element
-      this.observer.observe(element)
+      this.observeRef(value)
     })
 
-    const footer = (findNodeHandle(this.footer.current) as unknown) as Element
+    this.observeRef(this.footer)
+  }
+
+  observeRef = (ref: React.RefObject<View>) => {
```
Comment (aaronmgdr) [inline packages/web/src/brandkit/common/Page.tsx:208]: moves previously duplicated code into one spot

File: packages/web/src/shared/Button.3.tsx [hunk 2]
```diff
@@ -157,7 +156,7 @@ interface Props {
   kind?: BTN
   children: React.ReactNode
   size?: SIZE
-  style?: TextStyle
+  style?: TextStyle | TextStyle[]
```
Comment (aaronmgdr) [inline packages/web/src/shared/Button.3.tsx:159]: was not typed right before

File: packages/web/src/shared/Footer.tsx [hunk 3]
```diff
@@ -64,35 +63,27 @@ const Social = React.memo(function _Social() {
           <MediumLogo color={colors.dark} height={height} />
         </View>
         <View style={styles.socialIcon}>
-          <Link href={CeloLinks.gitHub}>
-            <a>
-              <Octocat color={colors.dark} size={height} />
-            </a>
-          </Link>
+          <a target="_blank" rel="noopener" href={CeloLinks.gitHub}>
```
Comment (aaronmgdr) [inline packages/web/src/shared/Footer.tsx:66]: this was throwing warnings for a while now, glad to have finally found out where they came from

File: packages/web/src/react-native-aug.d.ts [hunk 4]
```diff
@@ -10,29 +10,34 @@ declare module 'react-native' {
     gridArea?: string
   }
 
+  interface KeyFrame {
+    [string]: ViewStyle
+  }
+
   interface ViewStyle {
-    position?: 'absolute' | 'relative' | 'fixed' | 'static'
```
Comment (aaronmgdr) [inline packages/web/src/react-native-aug.d.ts:14]: alphabetize and add boxShadow

**Review by cmcewen — APPROVED** [review]

**Pull request merged**